src/components/ObsDetails/DetailsTab/DetailsTab.js

```diff
@@ -138,7 +138,12 @@ const DetailsTab = ( { currentUser, observation }: Props ): Node => {
         <DateDisplay
           classNameMargin="mb-[12px]"
           label={t( "Date-observed-header-short" )}
-          dateString={checkCamelAndSnakeCase( observation, "timeObservedAt" )}
+          dateString={
+            checkCamelAndSnakeCase( observation, "timeObservedAt" )
+            || checkCamelAndSnakeCase( observation, "observed_on_string" )
```

Comment:
checkCamelAndSnakeCase checks if a snake_case version of a camelCase key is present but not vice versa. Maybe it would be a bit less confusing to access the key directly here instead.